js/render/painter/use_program.js

```diff
@@ -61,26 +61,26 @@ var definitions = {
     }
 };
 
-module.exports._createProgram = function(name, macros) {
+var vertexUtilSource = fs.readFileSync(path.join(__dirname, '../../../shaders/util.vertex.glsl'), 'utf8');
+
+module.exports._createProgram = function(name, defines, vertexPragmas, fragmentPragmas) {
     var gl = this.gl;
     var program = gl.createProgram();
     var definition = definitions[name];
 
-    var defines = '';
-    if (macros) {
-        for (var m = 0; m < macros.length; m++) {
-            defines += '#define ' + macros[m] + '\n';
-        }
+    var definesSource = '';
+    for (var j = 0; definesSource && j < definesSource.length; j++) {
```

Comment:
`definesSource && j < definesSource.length` is always false, because `definesSource` is an empty string. Is this intended?